src/wattsi.pas

```diff
@@ -1015,24 +1015,34 @@    TCrossReferences = record
 
       MDNBox := E(eAside);
 
-      if (Element.HasProperties(propHeading)) then
+      if (Element.HasProperties(propHeading) or
+         HasAncestor(Element, nsHTML, eH2) or
+         HasAncestor(Element, nsHTML, eH3) or
+         HasAncestor(Element, nsHTML, eH4) or
+         HasAncestor(Element, nsHTML, eH5) or
+         HasAncestor(Element, nsHTML, eH6)) then
```

Comment:
It might be nicer, and slightly faster, to introduce a HasAncestorWithProperties function, or maybe just a HasAncestor function. Doing 5 ancestor traversals per element seems likely to slow things down a decent bit. (But maybe it won't matter much because our tree is generally shallow?)

Reply:
> It might be nicer, and slightly faster, to introduce a HasAncestorWithProperties function

Yep, agreed  — so added